### PR TITLE
mercurial: clean untracked files

### PR DESCRIPTION
**Pull request by philpep**

### CODE REVIEW

File: asv/plugins/mercurial.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def checkout_existing():
             with hglib.open(path) as subrepo:
                 subrepo.pull()
                 subrepo.update(commit_hash, clean=True)
```
Comment (mdboom) [inline asv/plugins/mercurial.py]: Ah -- I see.  `clean=True` here only discards uncommitted changes, not untracked files.  This makes sense, then, and brings the behavior inline with the `git` plugin.

**Comment by mdboom** [comment]
It looks like this functionality needs a mercurial plugin.  Will need to add that to `.travis.yml` and update requirement docs etc.

**Comment by pv** [comment]
It would be better to somehow tell Mercurial to enable the plugin here.
Users generally may not have the `purge` plugin enabled.

**Comment by philpep** [comment]
I added explicit enabling of the purge extension here.

**Comment by pv** [comment]
Thanks

**Pull request merged**